packages/x-date-pickers/src/TimePicker/TimePicker.types.ts

```diff
@@ -50,4 +50,8 @@ export interface TimePickerProps<TDate>
    * @default {}
    */
   slotProps?: TimePickerSlotsComponentsProps<TDate>;
+  /**
+   * The `name` attribute of the picker.
+   */
+  name?: string;
```

Comment:
This prop should probably not be defined on these interfaces because they are also applied to the static variants which have no input.

The best interface is probably `BaseNonRangeNonStaticPickerProps` which targets `TimePicker` `DatePicker and `DateTimePicker` (and there mobile / desktop variants).

Or `BaseNonStaticPickerProps` which also targets `DateRangePicker`.
This would not work on the default `DateRangePicker` which has 2 inputs but we could wire it for its [single input version](https://mui.com/x/react-date-pickers/date-range-picker/#use-a-single-input-field).
@LukasTy not sure what your opinion is here

____

And we should probably make sure that the behaviors you are trying to achieve with the `name` prop works with https://github.com/mui/mui-x/pull/10649

Reply:
A good solution could be to add it to `BaseNonRangeNonStaticPickerProps` and `BaseRangeNonStaticPickerProps` like we did for `inputRef` and `label`.
That way we can set a better JSDoc on the range components to add "Ignored if the field has several inputs."

Reply:
> A good solution could be to add it to BaseNonRangeNonStaticPickerProps and BaseRangeNonStaticPickerProps like we did for inputRef and label.

Let's go for this solution then, it's the most coherent one if we look at similar props.